src/Core/Services/Implementations/PolicyService.cs

```diff
@@ -132,6 +137,42 @@ public class PolicyService : IPolicyService
         await _eventService.LogPolicyEventAsync(policy, Enums.EventType.Policy_Updated);
     }
 
+    public async Task<ICollection<OrganizationUserPolicyDetails>> GetPoliciesApplicableToUserAsync(Guid userId, PolicyType policyType, OrganizationUserStatusType minStatus = OrganizationUserStatusType.Accepted)
+    {
+        var result = await QueryOrganizationUserPolicyDetailsAsync(userId, policyType, minStatus);
+        return result.ToList();
+    }
+
+    public async Task<bool> AnyPoliciesApplicableToUserAsync(Guid userId, PolicyType policyType, OrganizationUserStatusType minStatus = OrganizationUserStatusType.Accepted)
+    {
+        var result = await QueryOrganizationUserPolicyDetailsAsync(userId, policyType, minStatus);
+        return result.Any();
+    }
+
+    private async Task<IEnumerable<OrganizationUserPolicyDetails>> QueryOrganizationUserPolicyDetailsAsync(Guid userId, PolicyType policyType, OrganizationUserStatusType minStatus = OrganizationUserStatusType.Accepted)
+    {
+        var organizationUserPolicyDetails = await _organizationUserRepository.GetByUserIdWithPolicyDetailsAsync(userId);
+        var minUserType = GetMinimumOrganizationUserTypeForPolicyType(policyType);
+        return organizationUserPolicyDetails.Where(o =>
+            o.PolicyType == policyType &&
+            o.PolicyEnabled &&
+            o.OrganizationUserType >= minUserType &&
+            o.OrganizationUserStatus >= minStatus &&
+            !o.IsProvider);
+    }
+
+    private OrganizationUserType GetMinimumOrganizationUserTypeForPolicyType(PolicyType policyType)
+    {
+        switch (policyType)
+        {
+            case PolicyType.RequireSso:
```

Comment:
@shane-melton pointed out in #2714 that the Master Password policy applies to Owners and Admins, so that needs to be handled here as an exception to the general rule that Owners and Admins are excluded.

I also noticed that this method doesn't really work because `Owner` is greater than `User`, so if `minUserType == User`, it'll also be applied to owners. Really it needs to be a `maxUserType`, but that's confusing.

Maybe instead of thinking of minimum/maximum types, it could be `private OrganizationUserType[] GetUserTypesExcludedFromPolicy(PolicyType policyType)` or something, or even `private bool UserTypeExcludedFromPolicy(PolicyType policyType, User user)`.

Reply:
> I also noticed that this method doesn't really work because Owner is greater than User, so if minUserType == User, it'll also be applied to owners. Really it needs to be a maxUserType, but that's confusing.

Owner is inferior than User:
```
Owner = 0
User = 2
```

I think this would work:
```suggestion
            case PolicyType.MasterPassword:
                return OrganizationUserType.Owner;
            case PolicyType.RequireSso:
```

Reply:
I thought a bit about this and your suggestion made sense to me. I've implemented `private OrganizationUserType[] GetUserTypesExcludedFromPolicy(PolicyType policyType)`

Reply:
ah, sorry I must have been thinking about statuses. I thought a higher enum value == higher permissions. But I think that shows why this is a better abstraction anyway 😁